paper-server/patches/sources/net/minecraft/server/MinecraftServer.java.patch

```diff
@@ -1110,11 +1110,33 @@
 -            this.worldData.setDifficulty(this.worldData.isHardcore() ? Difficulty.HARD : difficulty);
 -            this.updateMobSpawningFlags();
 -            this.getPlayerList().getPlayers().forEach(this::sendDifficultyUpdate);
-+    // Paper start - per level difficulty
-+    public void setDifficulty(ServerLevel level, Difficulty difficulty, boolean forced) {
++    // Paper start - per level difficulty, WorldDifficultyChangeEvent
```

Comment:
I feel this things can be moved to the CraftBukkit side

Reply:
This should works like the getter/setter in CraftWorld